diff --git a/index.js b/index.js
index 23384ea16bb848616886f356d8d69f94325535bf..51ba4af51d075d0e4cfe9d38e57ab0f55ed47686 100644
--- a/index.js
+++ b/index.js
@@ -126,6 +126,11 @@ class Extractor {
 
     if (symlink) {
       const link = await getStream(readStream)
+      const target = path.resolve(path.dirname(dest), link)
+      const relativeTarget = path.relative(this.opts.dir, target)
+      if (relativeTarget.split(path.sep).includes('..') || path.isAbsolute(relativeTarget)) {
+        throw new Error(`Out of bound symlink target "${link}" found while processing file ${entry.fileName}`)
+      }
       debug('creating symlink', link, dest)
       await fs.symlink(link, dest)
     } else {
